failed to use loader on windows

```
> rebem-starter-kit@0.0.0 start D:\wspace\starter-kit
> start-runner ./tasks.js "dev"

→ env: start
→ env: NODE_ENV = development
→ env: done

→ webpack dev: start
→ webpack dev: http://localhost:3000/webpack-dev-server/
→ webpack dev: done
Time: 1491ms
chunk    {0} bundle.js (main) 644 kB [rendered]

ERROR in ./src/index.js
Module not found: Error: Cannot resolve module 'D:wspacestarter-kitsrccomponentsappindex.js' in D:\wspace\starter-kit\src
 @ ./src/index.js 5:53-113

ERROR in ./src/index.js
Module not found: Error: Cannot resolve module 'D:wspacestarter-kit
ebem-theme-reseuildappstyles.css' in D:\wspace\starter-kit\src
 @ ./src/index.js 5:127-211

ERROR in ./src/index.js
Module not found: Error: Cannot resolve module 'D:wspacestarter-kitsrc  heme-bootstrapappstyles.less' in D:\wspace\starter-kit\src
 @ ./src/index.js 5:213-281

ERROR in ./src/index.js
Module not found: Error: Cannot resolve module 'D:wspacestarter-kitsrccomponentsappstyles.less' in D:\wspace\starter-kit\src
 @ ./src/index.js 5:283-346
webpack: bundle is now VALID.
```

runner looks ok:

``` js
    var possibleMainPaths = getPossibleMainPaths(currentPath, required.component, layers);
    var possibleStylesPaths = getPossibleStylesPaths(required.component, layers);
    console.log(possibleMainPaths, possibleStylesPaths)
```

```
→ env: start
→ env: NODE_ENV = development
→ env: done

→ webpack dev: start
→ webpack dev: http://localhost:3000/webpack-dev-server/
→ webpack dev: done
[ 'D:\\wspace\\starter-kit\\node_modules\\rebem-core-components\\build\\app\\index.js',
  'D:\\wspace\\starter-kit\\src\\theme-bootstrap\\app\\index.js',
  'D:\\wspace\\starter-kit\\src\\components\\app\\index.js' ] [ 'D:\\wspace\\starter-kit\\node_modules\\rebem-theme-reset\\build\\app\\styles.css',
  'D:\\wspace\\starter-kit\\src\\theme-bootstrap\\app\\styles.less',
  'D:\\wspace\\starter-kit\\src\\components\\app\\styles.less' ]
Time: 1033ms
chunk    {0} bundle.js (main) 644 kB [rendered]

ERROR in ./src/index.js
Module not found: Error: Cannot resolve module 'D:wspacestarter-kitsrccomponentsappindex.js' in D:\wspace\starter-kit\src
 @ ./src/index.js 5:53-113

ERROR in ./src/index.js
Module not found: Error: Cannot resolve module 'D:wspacestarter-kit
ebem-theme-reseuildappstyles.css' in D:\wspace\starter-kit\src
 @ ./src/index.js 5:127-211

ERROR in ./src/index.js
Module not found: Error: Cannot resolve module 'D:wspacestarter-kitsrc  heme-bootstrapappstyles.less' in D:\wspace\starter-kit\src
 @ ./src/index.js 5:213-281

ERROR in ./src/index.js
Module not found: Error: Cannot resolve module 'D:wspacestarter-kitsrccomponentsappstyles.less' in D:\wspace\starter-kit\src
 @ ./src/index.js 5:283-346
webpack: bundle is now VALID.
```

all pathes properly escaped.

```
./src/index.js
Module not found: Error: Cannot resolve module 'D:wspacestarter-kitsrccomponentsappindex.js' in D:\wspace\starter-kit\src
resolve module D:wspacestarter-kitsrccomponentsappindex.js in D:\wspace\starter-kit\src
  looking for modules in D:\wspace\starter-kit\node_modules
    D:\wspace\starter-kit\node_modules\D:wspacestarter-kitsrccomponentsappindex.js doesn't exist (module as directory)
    resolve 'file' D:wspacestarter-kitsrccomponentsappindex.js in D:\wspace\starter-kit\node_modules
      resolve file
        D:\wspace\starter-kit\node_modules\D:wspacestarter-kitsrccomponentsappindex.js doesn't exist
        D:\wspace\starter-kit\node_modules\D:wspacestarter-kitsrccomponentsappindex.js.webpack.js doesn't exist
        D:\wspace\starter-kit\node_modules\D:wspacestarter-kitsrccomponentsappindex.js.web.js doesn't exist
        D:\wspace\starter-kit\node_modules\D:wspacestarter-kitsrccomponentsappindex.js.js doesn't exist
        D:\wspace\starter-kit\node_modules\D:wspacestarter-kitsrccomponentsappindex.js.json doesn't exist
[D:\wspace\starter-kit\node_modules\D:wspacestarter-kitsrccomponentsappindex.js]
[D:\wspace\starter-kit\node_modules\D:wspacestarter-kitsrccomponentsappindex.js]
[D:\wspace\starter-kit\node_modules\D:wspacestarter-kitsrccomponentsappindex.js.webpack.js]
[D:\wspace\starter-kit\node_modules\D:wspacestarter-kitsrccomponentsappindex.js.web.js]
[D:\wspace\starter-kit\node_modules\D:wspacestarter-kitsrccomponentsappindex.js.js]
[D:\wspace\starter-kit\node_modules\D:wspacestarter-kitsrccomponentsappindex.js.json]
 @ ./src/index.js 5:53-113
```

--- a/lib/index.js
+++ b/lib/index.js
@@ -10,7 +10,7 @@ export default function(source) {
     const callback = this.async();
     const options = loaderUtils.parseQuery(this.query);
 
-    transform(source, this.resourcePath, options)
+    transform.call(this, source, this.resourcePath, options)
         .then(result => {
             callback(null, result);
         })

--- a/lib/transform.js
+++ b/lib/transform.js
@@ -3,6 +3,7 @@ import falafel from 'falafel';
 import getResults from './runner';
 import getOutput from './output';
 import getOutputWithInjects from './inject/output';
+import { stringifyRequest } from 'loader-utils';
 
 export default function(source, file, options) {
     const consumerPaths = options.layers.map(layer => layer.path).concat(options.consumers || []);
@@ -26,6 +27,13 @@ export default function(source, file, options) {
 
             requiresPromises.push(
                 getResults(rawRequireValue, file, options.layers).then(result => {
+                    if (result.component) {
+                        result.component = stringifyRequest(this, result.component).replace(/"/g, '');
+                    }
+                    result.styles = result.styles.map((style) => {
+                        return stringifyRequest(this, style).replace(/"/g, '');
+                    });
+
                     if (options.injectable) {
                         node.update(getOutputWithInjects(result, options));
                     } else {

--- a/test/fixtures/?class/import-factory/result.js
+++ b/test/fixtures/?class/import-factory/result.js
@@ -1,1 +1,1 @@
-/* #target */ require('layer-0/target/index.js')['default'];
+/* #target */ require('./layer-0/target/index.js')['default'];

--- a/test/fixtures/?class/simple/result.js
+++ b/test/fixtures/?class/simple/result.js
@@ -1,1 +1,1 @@
-/* #target?class */ require('layer-0/target/index.js')['default'];
+/* #target?class */ require('./layer-0/target/index.js')['default'];

--- a/test/fixtures/?class/with-styles/result.js
+++ b/test/fixtures/?class/with-styles/result.js
@@ -1,1 +1,1 @@
-/* #target?class */ require('layer-0/target/index.js')['default']; require('layer-0/target/styles.less');
+/* #target?class */ require('./layer-0/target/index.js')['default']; require('./layer-0/target/styles.less');

--- a/test/fixtures/?inject/?class/result.js
+++ b/test/fixtures/?inject/?class/result.js
@@ -1,1 +1,1 @@
-/* #target?class&inject */ function(injects) { return require('rebem-layers-loader/build/inject/loader!layer-0/target/index.js')['default'](injects); };
+/* #target?class&inject */ function(injects) { return require('rebem-layers-loader/build/inject/loader!./layer-0/target/index.js')['default'](injects); };

--- a/test/fixtures/?inject/?styles/result.js
+++ b/test/fixtures/?inject/?styles/result.js
@@ -1,1 +1,1 @@
-/* #target?styles&inject */ require('layer-0/target/styles.less');
+/* #target?styles&inject */ require('./layer-0/target/styles.less');

--- a/test/fixtures/?inject/no-injects/import-factory/result.js
+++ b/test/fixtures/?inject/no-injects/import-factory/result.js
@@ -1,1 +1,1 @@
-/* #target */ (function() { module.__rebem__ = module.__rebem__ || {}; module.__rebem__['layer-0/target/index.js'] = '#target'; var out = require('layer-0/target/index.js')['default'];  return out; })();
+/* #target */ (function() { module.__rebem__ = module.__rebem__ || {}; module.__rebem__['./layer-0/target/index.js'] = '#target'; var out = require('./layer-0/target/index.js')['default'];  return out; })();

--- a/test/fixtures/?inject/no-injects/simple/result.js
+++ b/test/fixtures/?inject/no-injects/simple/result.js
@@ -1,1 +1,1 @@
-/* #target */ (function() { module.__rebem__ = module.__rebem__ || {}; module.__rebem__['layer-0/target/index.js'] = '#target'; var out = require('react').createFactory(require('layer-0/target/index.js')['default']); return out; })();
+/* #target */ (function() { module.__rebem__ = module.__rebem__ || {}; module.__rebem__['./layer-0/target/index.js'] = '#target'; var out = require('react').createFactory(require('./layer-0/target/index.js')['default']); return out; })();

--- a/test/fixtures/?inject/simple/result.js
+++ b/test/fixtures/?inject/simple/result.js
@@ -1,1 +1,1 @@
-/* #target?inject */ function(injects) { return require('react').createFactory(require('rebem-layers-loader/build/inject/loader!layer-0/target/index.js')['default'](injects)); };
+/* #target?inject */ function(injects) { return require('react').createFactory(require('rebem-layers-loader/build/inject/loader!./layer-0/target/index.js')['default'](injects)); };

--- a/test/fixtures/?inject/with-styles/result.js
+++ b/test/fixtures/?inject/with-styles/result.js
@@ -1,1 +1,1 @@
-/* #target?inject */ function(injects) { return require('react').createFactory(require('rebem-layers-loader/build/inject/loader!layer-0/target/index.js')['default'](injects)); }; require('layer-0/target/styles.less');
+/* #target?inject */ function(injects) { return require('react').createFactory(require('rebem-layers-loader/build/inject/loader!./layer-0/target/index.js')['default'](injects)); }; require('./layer-0/target/styles.less');

--- a/test/fixtures/?styles/simple/result.js
+++ b/test/fixtures/?styles/simple/result.js
@@ -1,1 +1,1 @@
-/* #target?styles */ require('layer-0/target/styles.less');
+/* #target?styles */ require('./layer-0/target/styles.less');

--- a/test/fixtures/simple/cache/result.js
+++ b/test/fixtures/simple/cache/result.js
@@ -1,2 +1,2 @@
-/* #target */ require('react').createFactory(require('layer-0/target/index.js')['default']);
-/* #target */ require('react').createFactory(require('layer-0/target/index.js')['default']);
+/* #target */ require('react').createFactory(require('./layer-0/target/index.js')['default']);
+/* #target */ require('react').createFactory(require('./layer-0/target/index.js')['default']);

--- a/test/fixtures/simple/multiple-layers/middle/result.js
+++ b/test/fixtures/simple/multiple-layers/middle/result.js
@@ -1,1 +1,1 @@
-/* #target */ require('react').createFactory(require('layer-2/target/index.js')['default']);
+/* #target */ require('react').createFactory(require('./layer-2/target/index.js')['default']);

--- a/test/fixtures/simple/multiple-layers/nearest/result.js
+++ b/test/fixtures/simple/multiple-layers/nearest/result.js
@@ -1,1 +1,1 @@
-/* #target */ require('react').createFactory(require('layer-1/target/index.js')['default']);
+/* #target */ require('react').createFactory(require('./layer-1/target/index.js')['default']);

--- a/test/fixtures/simple/multiple-layers/self-middle/result.js
+++ b/test/fixtures/simple/multiple-layers/self-middle/result.js
@@ -1,1 +1,1 @@
-/* #target */ require('react').createFactory(require('layer-0/target/index.js')['default']);
+/* #target */ require('react').createFactory(require('./layer-0/target/index.js')['default']);

--- a/test/fixtures/simple/multiple-layers/self/result.js
+++ b/test/fixtures/simple/multiple-layers/self/result.js
@@ -1,1 +1,1 @@
-/* #target */ require('react').createFactory(require('layer-0/target/index.js')['default']);
+/* #target */ require('react').createFactory(require('./layer-0/target/index.js')['default']);

--- a/test/fixtures/simple/multiple-layers/through/result.js
+++ b/test/fixtures/simple/multiple-layers/through/result.js
@@ -1,1 +1,1 @@
-/* #target */ require('react').createFactory(require('layer-0/target/index.js')['default']);
+/* #target */ require('react').createFactory(require('./layer-0/target/index.js')['default']);

--- a/test/fixtures/simple/single-layer/result.js
+++ b/test/fixtures/simple/single-layer/result.js
@@ -1,1 +1,1 @@
-/* #target */ require('react').createFactory(require('layer-0/target/index.js')['default']);
+/* #target */ require('react').createFactory(require('./layer-0/target/index.js')['default']);

--- a/test/fixtures/with-styles/multiple-layers/result.js
+++ b/test/fixtures/with-styles/multiple-layers/result.js
@@ -1,1 +1,1 @@
-/* #target */ require('react').createFactory(require('layer-1/target/index.js')['default']); require('layer-0/target/styles.less'); require('layer-1/target/styles.less');
+/* #target */ require('react').createFactory(require('./layer-1/target/index.js')['default']); require('./layer-0/target/styles.less'); require('./layer-1/target/styles.less');

--- a/test/fixtures/with-styles/single-layer/result.js
+++ b/test/fixtures/with-styles/single-layer/result.js
@@ -1,1 +1,1 @@
-/* #target */ require('react').createFactory(require('layer-0/target/index.js')['default']); require('layer-0/target/styles.less');
+/* #target */ require('react').createFactory(require('./layer-0/target/index.js')['default']); require('./layer-0/target/styles.less');

--- a/test/helpers/index.js
+++ b/test/helpers/index.js
@@ -10,6 +10,9 @@ export default function(params) {
 
     const transform = require('../../lib/transform');
     const testPath = path.resolve('./test/fixtures/', params.path);
+    const loaderContext = {
+        context: testPath
+    };
     const componentPath = path.resolve(testPath, params.test);
     const resultPath = path.resolve(testPath, 'result.js');
     const options = {
@@ -24,7 +27,7 @@ export default function(params) {
 
     return readFile(componentPath, 'utf-8').then(sourceData => {
         return readFile(resultPath, 'utf-8').then(resultData => {
-            return transform(sourceData, componentPath, options).then(result => {
+            return transform.call(loaderContext, sourceData, componentPath, options).then(result => {
                 result = result.split(testPath + '/').join('');
 
                 if (result !== resultData) {

--- a/test/helpers/index.js
+++ b/test/helpers/index.js
@@ -28,8 +28,6 @@ export default function(params) {
     return readFile(componentPath, 'utf-8').then(sourceData => {
         return readFile(resultPath, 'utf-8').then(resultData => {
             return transform.call(loaderContext, sourceData, componentPath, options).then(result => {
-                result = result.split(testPath + '/').join('');
-
                 if (result !== resultData) {
                     console.log('actual:', result);
                     console.log('expected:', resultData);

